### ISSUE TITLE
Introducing periodic topology mechanism for JedisCluster

### ISSUE THREAD

**Issue by yangbodong22011** [redis/jedis#3595]
The current JedisCluster strategy for reacquiring routing tables (calling `Cluster Slots`) is:

1. When the client encounters the server, it returns `MOVED`.
2. When the client fails for `maxAttempts` times.

One of our customers, who was using JedisCluster but in a quiescent state (i.e. without any access), initially connected to cluster A, but then some of the nodes in cluster A were taken offline and the IPs were reused in cluster B. At this time JedisCluster is connected to cluster B.

### How to Fix

We need to add a periodic topology refresh mechanism to JedisCluster. For example, Lettuce's `enablePeriodicRefresh` [parameter](https://github.com/lettuce-io/lettuce-core/wiki/Client-options#cluster-specific-options).
```
ClusterTopologyRefreshOptions refreshOptions = ClusterTopologyRefreshOptions.builder()
    .enablePeriodicRefresh(Duration.ofSeconds(15)).build();
```

**Comment by sazzad16** [comment]
:+1:

**Comment by brunogbv** [comment]
bumping this as we had a issue that relates to this: https://github.com/redis/jedis/issues/2550#issuecomment-1779679240

**Comment by sazzad16** [comment]
Resolved by #3596
